packages/toolchain/src/app/components/SpaceDropdownPageSection.tsx

```diff
@@ -44,7 +44,8 @@ export class SpaceDropdownPageSection extends React.Component<Props> {
             {username != null && (
               <DropdownItemRoute
                 key="all"
-                href={`/${username}/${NO_SPACE_PATH}${subPath ? `/${subPath}` : ''}`}
+                // if there is no subPath we must go to the root
+                href={subPath ? `/${username}/${NO_SPACE_PATH}/${subPath}` : ''}
```

Comment:
Instead of ```username```, it should be space owner username. Otherwise, it will give 404 error If the user visits the other space. where the user is the collaborator of that space.

Reply:
This dropdown only deals with the current user's spaces and not spaces from other users.

Reply:
Isn't this is the same dropdrown that is visible on secondary header of the dropdown.

Reply:
It is. But that header is only showing the logged in users's spaces. Perhaps it should show all spaces the user collaborates with. But that is a feature change. I'll talk to UX as the intent from the invision wasn't clear.

Reply:
My mistake. You're right, the api does return all spaces the user collaborates with. Will address.

Reply:
Yes, it shows all the space the users collaborates with.